lib/perl/Genome/Model/Tools/DetectVariants2/Dispatcher.pm

```diff
@@ -300,18 +303,25 @@ sub _dump_dv_cmd {
     return 1;
 }
 
-sub get_relative_path_to_output_directory {
+sub get_relative_path_to_result_directory {
     my $self = shift;
-    my $full_path = shift;
-    my $relative_path = $full_path;
-    my $temp_path = $self->_temp_staging_directory;
-    $relative_path =~ s/$temp_path\/?//;
-    return $relative_path;
+    my $result_path = shift;
+
+    my $relative_path = $result_path;
+
+    my $dispatcher_path = $self->output_directory;
+    if ($relative_path =~ /$dispatcher_path\/?/ ) {
+        $relative_path =~ s/$dispatcher_path\/?//;
+        return $relative_path;
+    }
+
+    return;
```

Comment:
I just wasn't thinking right... it's actually looking for the path followed by 1 or 0 slashes.  It's OK after all.  If you want to be extra super duper safe, it probably should be matching a (again, partial?) path in its entirety.  Meaning if $dispatcher path is "foo", it should correctly match and snip out
* /path/to/foo
* /path/to/foo/bar

while leaving
* /path/to/foobar
* /blah/foobar/baz
alone.

In practice, I'm sure it won't come up, but a more complicated regex could handle it.  Maybe something like ```s#(?<=/)$dispatcher_path(?:/|\Z)##;```  That looks for a slash, then $dispatcher_path, then either a slash or the end of the string, but the first slash isn't cut out because it's a look-behind assertion.